app/services/site_setting/update.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class SiteSetting::Update
+  include Service::Base
+
+  policy :current_user_is_admin
+  contract do
+    attribute :setting_name
+    attribute :new_value
+    attribute :allow_changing_hidden, :boolean, default: false
+
+    before_validation do
+      self.setting_name = setting_name&.to_sym
+      self.new_value = new_value.to_s.strip
+    end
+
+    validates :setting_name, presence: true
+
+    after_validation do
+      next if setting_name.blank?
+      self.new_value =
```

Comment:
Moving all the processing from a dedicated step to the contract, as it’s one of its responsibilities. We could probably extract this even more to a `ActiveModel::Type::Value` or something.